### PR TITLE
Set up Jest

### PR DESCRIPTION
**Pull request by ryotogashi**

### CODE REVIEW

**Comment by ryotogashi** [comment]
#3

**Comment by ryotogashi** [comment]
@axross Can you check?👁

File: __tests__/pages/index.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import * as renderer from 'react-test-renderer';
+
+import Home from '../../pages/index';
```
Comment (axross) [inline __tests__/pages/index.spec.tsx]:
```suggestion
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Home from '../../pages/index';
```

File: __tests__/pages/index.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+describe('Home test', () => {
+  it('tests home page render correctly', () => {
+    const home = renderer.create(<Home />).toJSON();
+
+    expect(home).toMatchSnapshot();
```
Comment (axross) [inline __tests__/pages/index.spec.tsx]:
```suggestion
    const renderer = TestRenderer.create(<Home />);

    expect(renderer.toJSON()).toMatchSnapshot();
```

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    'ts-jest': {
+      tsConfig: {
+        ...require('./tsconfig.json').compilerOptions,
+        jsx: 'react'
+      }
+    }
+  },
+  testMatch: ['**/__tests__/**/*.spec.+(ts|tsx|js)'],
+  preset: 'ts-jest'
+};
```
Comment (axross) [inline jest.config.js]: Can you try using `ts-jest`'s default setting as much as possible?

I guess you don't need to do anything apart from TypeScript's compiler option `jsx: "react"`.

File: package.json [hunk 4]
```diff
@@ -15,16 +15,22 @@
...
+    "react-test-renderer": "^16.13.0",
+    "ts-jest": "^25.2.1",
     "typescript": "^3.8.3"
   },
   "scripts": {
     "dev": "next dev",
     "build": "next build",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "jest",
+    "test:watch": "jest --watch"
```
Comment (axross) [inline package.json]: There's a lot of variety to run tests, such as `--watch`, `--notify`, `--updateSnapshots`. So we should put options manually such like runing `npm test -- --watch` or (with `--notify`) when we want to test code in watch mode.

```suggestion
```

File: __tests__/pages/index.spec.tsx (hunk 1 above)
Comment (ryotogashi) [reply __tests__/pages/index.spec.tsx]: @axross oh I see, It was my intention. I wanted to separate import from Library and our files but I ok!

**Comment by ryotogashi** [comment]
@axross Thx! I've fixed.

**Pull request merged**